src/platform/plugins/shared/unified_doc_viewer/public/components/doc_viewer_traces_overview/sub_components/trace_summary.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import { TRANSACTION_NAME_FIELD } from '@kbn/discover-utils';
+import { EuiHorizontalRule } from '@elastic/eui';
+import React, { useState, useEffect } from 'react';
+import { useTransactionContext } from '../hooks/use_transaction';
+import { FieldWithActions } from './field_with_actions/field_with_actions';
+import { FieldConfiguration } from '../resources/get_field_configuration';
+
+export interface TraceSummaryProps {
+  fieldId: string;
+  fieldConfiguration: FieldConfiguration;
+}
+
+export function TraceSummary({ fieldConfiguration, fieldId }: TraceSummaryProps) {
+  const { transaction, loading } = useTransactionContext();
+  const [fieldValue, setFieldValue] = useState(fieldConfiguration.value);
+  const isTransactionNameField = fieldId === TRANSACTION_NAME_FIELD;
+  const displayLoading = isTransactionNameField && !fieldValue;
+
+  useEffect(() => {
+    if (isTransactionNameField && !fieldValue && transaction && !loading) {
+      setFieldValue(transaction.name);
+    }
+  }, [transaction, fieldId, loading, fieldValue, isTransactionNameField]);
```

Comment:
Better to have on transactin.name as a dependency to avoid unnecessary renders.

```suggestion
  useEffect(() => {
    if (isTransactionNameField && !fieldValue && transaction?.nane && !loading) {
      setFieldValue(transaction.name);
    }
  }, [transaction?.nam, fieldId, loading, fieldValue, isTransactionNameField]);
```

Reply:
Why do you need the `fieldId` defined in the deps?